better_performance_on_the_client.md

```diff
@@ -0,0 +1,32 @@
+# Improving Application Performance and Responsiveness on the Client
+Performance and responsiveness on client side applications are very important topics which usually don't receive much attention until they are an obvious problem. At Ubiqua, as we began to increment the number of features our product offered, we started to experience the following problems:
+
+- Application reloaded too often in mobile devices.
+- Application took too long to reload.
+- UI thread performed heavy computations which resulted in bad UX.
+- Overall application performance was poor.
+
+In this blogpost, I'll show some of the techniques we used to enhance the performance and responsiveness of our application.
+
+## Measure Then Act
+The very first thing we have to make sure we do before attempting to fix anything, is to measure what we believe is causing the application to perform poorly. Chrome DevTools offer a variety of tools which facilitate this:
+
+- [Collect JavaScript CPU Profile](https://developer.chrome.com/devtools/docs/cpu-profiling): CPU profiles help to visualize where the execution time is spent in the JavaScript that is being run. Collecting JavaScript CPU profile was critical for us to replicate user actions we already knew were performing slow. We identified methods executed during those user actions which were taking too long and then optimized them. After these methods were optimized, we collected another CPU profile by executing the same previous user actions and we measured if the execution time had decreased in comparison with the first measurements.
```

Comment:
Should probably mention the console API method ``console.profile([label])`` to measure user actions from specific places in the code.